### PR TITLE
PSFR-1213 Enable resuming a report entry

### PR DESCRIPTION
**Pull request by mkerry-moj**
* Cache against userid instead of sessionid
* Increase ttl of keys to 5 days (up from until midnight of current day)
* When moving from task list into a pathway, check if in progress and move to the next question based on previous responses if it is
* Change Not started on task list page to Incomplete

### CODE REVIEW

File: server/data/assessmentStore.ts [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
+import type { RedisClient } from './redisClient'
 import logger from '../../logger'
 import { AssessmentPage, SubmittedInput } from './model/immediateNeedsReport'
-import { secondsUntilMidnight } from '../utils/utils'
 
-const assessmentPrefix = 'assessment:'
-const answeredQuestionsPrefix = 'answered:'
-const currentPagePrefix = 'currentPage:'
-const editedQuestionPrefix = 'edit:'
+const defaultTimeToLive = hoursToSeconds(24 * 5)
```
Comment (riccardo-noviello-moj) [inline server/data/assessmentStore.ts:7]: you could make this configurable in case Nick wants to test it in DEV with a shorter span

Comment (mkerry-moj) [reply server/data/assessmentStore.ts:7]: That's a good idea, thanks

File: server/routes/immediate-needs-report/immediateNeedsReportController.ts [hunk 2]
```diff
@@ -26,24 +26,29 @@ export default class ImmediateNeedsReportController {
...
         pathway,
       }
 
-      // Reset the cache at the point as starting new journey through the form
-      await this.assessmentStateService.reset(stateKey, pathway)
+      const existingInput = await this.assessmentStateService.prepareSubmission(stateKey)
+      const { questionsAndAnswers } = existingInput
+      let currentPageId = null
+
+      if (questionsAndAnswers.length > 0) {
```
Comment (riccardo-noviello-moj) [inline server/routes/immediate-needs-report/immediateNeedsReportController.ts:37]: `if (questionsAndAnswers && questionsAndAnswers.length > 0) {` for safety?

Comment (mkerry-moj) [reply server/routes/immediate-needs-report/immediateNeedsReportController.ts:37]: the state service has got `?? : {questionsAndAnswers: []}` in it

**Review by riccardo-noviello-moj — APPROVED** [review]

**Review by riccardo-noviello-moj — DISMISSED** [review]

**Review by riccardo-noviello-moj — APPROVED** [review]

**Pull request merged**